### PR TITLE
feature: added _AnalysisConfigGenerator for clarify

### PR DESCRIPTION
**Pull request by dosatos**
*Issue #, if available:*

*Description of changes:*
- Added `_AnalysisConfigGenerator` class to simplify maintainability of the config creation. 
    - Also, with this generator, customers may reproduce the state of the config locally during debugging (i.e. without a need to opening it in the S3 bucket)
- Simplified `job_name` creation part
- Added a private method `_last_analysis_config`) to store last analysis_config used for a run (to be useful in case of debugging)
- Extended the std out with the analysis config used for a run. Currently it already prints other values, like Job Name, Inputs, and Outputs:

```
Job Name:  Clarify-Bias-2022-07-22-09-00-31-294
Inputs:  [{...}]
Outputs:  [...}]
```

*Testing done:*
Unit Tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by larroy — COMMENTED** [review]
Two comments:

1) Can you please add types?
2) Can you please test a config including pre-post-explainability all at once?

Can the generator synthetise the config for all the methods enabled + explainability?

**Comment by dosatos** [comment]
> Can you please test a config including pre-post-explainability all at once?
> Can the generator synthetise the config for all the methods enabled + explainability?

So far there is no API to call for both Bias and Explainability at once. So I cannot. 
*(Maybe I did not get the question?)*

**Comment by larroy** [comment]
weird, but analysis config supports having a job with explainability and bias at once, how is it that we can't do it with the SDK? or am I missing something?

**Review by larroy — APPROVED** [review]
Changes look fine, one of my question is not addressed, but it should not block your PR. Thank you for your changes, LGTM.

File: src/sagemaker/clarify.py [hunk 1]
```diff
@@ -971,7 +973,7 @@ def _run(
     def run_pre_training_bias(
         self,
         data_config,
-        data_bias_config,
+        bias_config,
```
Comment (navinsoni) [inline src/sagemaker/clarify.py:976]: This change do not look backward compatible

File: src/sagemaker/clarify.py [hunk 2]
```diff
@@ -1041,7 +1042,7 @@ def run_pre_training_bias(
     def run_post_training_bias(
         self,
         data_config,
-        data_bias_config,
+        bias_config,
```
Comment (navinsoni) [inline src/sagemaker/clarify.py:1045]: same here

File: src/sagemaker/clarify.py (hunk 1 above)
Comment (dosatos) [reply src/sagemaker/clarify.py]: good point. let me revert it back

**Review by keerthanvasist — APPROVED** [review]
Changes LGTM

**Review by navinsoni — APPROVED** [review]

**Review by larroy — APPROVED** [review]
lgtm

**Comment by shreyapandit** [comment]
PR check for serverless inference failing because of known backend issue, will override and merge this to unblock team.

**Pull request merged**